devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusAnalytics.java

```diff
@@ -0,0 +1,79 @@
+package io.quarkus.gradle.tasks;
+
+import static io.quarkus.analytics.dto.segment.ContextBuilder.CommonSystemProperties.GRADLE_VERSION;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.inject.Inject;
+
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.TaskAction;
+
+import io.quarkus.analytics.AnalyticsService;
+import io.quarkus.analytics.config.FileLocationsImpl;
+import io.quarkus.analytics.dto.segment.TrackEventType;
+import io.quarkus.gradle.Prompter;
+
+public abstract class QuarkusAnalytics extends QuarkusTask {
+
+    private AnalyticsService analyticsService = AnalyticsService.getInstance(
```

Comment:
Not sure this is possible there, for all cases.
When we start devmode for the 1st time we need to pop the opt-in question and, if accepted, we then need to send analytics each time we start devmode and run the build.
As far as I can see the BuildWorker works only with the BuildTask... Will try that there.

Reply:
@snazy I tried to put the analytics on the build worker, but it's not executed in all builds, even with `--no-daemon --no-build-cache`

Reply:
In the end I decided Gradle build will only send analytics when changes are detected. On Maven, for now, it will send every time the build command is executed.

Reply:
IMO it's fine to only send build-analytics, when the build actually runs (no build -> no analytics).

Reply:
Yes, my only concern is that difference in behaviour between Gradle and Maven.